base/errorshow.jl

```diff
@@ -1104,7 +1104,29 @@ function fielderror_hint_handler(io, exc)
     end
 end
 
-Experimental.register_error_hint(fielderror_hint_handler, FieldError)
+Experimental.register_error_hint(fielderror_dict_hint_handler, FieldError)
+
+function fielderror_listfields_hint_handler(io, exc)
+    fields = fieldnames(exc.type)
+    if isempty(fields)
+        print(io, "; $(nameof(exc.type)) has no fields at all.")
+    else
+        print(io, ", available fields: $(join(map(k -> "`$k`", fields), ", "))")
+    end
+    props = _propertynames_bytype(exc.type)
+    isnothing(props) && return
+    props = setdiff(props, fields)
+    isempty(props) && return
+    print(io, "\nAvailable properties: $(join(map(k -> "`$k`", props), ", "))")
+end
+
+_extract_val(::Type{Val{V}}) where {V} = V
+_extract_val(@nospecialize _) = nothing
+_propertynames_bytype(::Type{T}) where {T} =
+    which(propertynames, (T,)) === which(propertynames, (Any,)) ? nothing :
+    _extract_val(promote_op(Val∘propertynames, T))
+
+Experimental.register_error_hint(fielderror_listfields_hint_handler, FieldError)
```

Comment:
New error hints should likely go in the `REPL` stdlib (where all the fuzzy string hints are located), not in Base

Reply:
> New error hints should likely go in the REPL stdlib (where all the fuzzy string hints are located), not in Base

This doesn't make sense to me. Keep them all in the same place until they are wholesale moved.